### PR TITLE
fix #4167 : Default integrationPlatform created at operator startup dose not honor OPERATOR_ID config

### LINKED ISSUE #4167: Default integrationPlatform created at operator startup dose not honor OPERATOR_ID config
See https://github.com/apache/camel-k/blob/eb9228597203f4c46fe2f488e59b7131826e1bde/pkg/cmd/operator/operator.go#L253-L267

If operator has not empty OPERATOR_ID, the default integrationPlatform created at startup should be annotated with:
`camel.apache.org/operator.id: $OPERATOR_ID` in order to be picked up for reconcilation.

### PR DESCRIPTION
**Pull request by valdar**
Fix #4167, if OPERATOR_ID configuration is used, the default `integrationPlatform` is now created with the correct annotation `camel.apache.org/operator.id: $OPERATOR_ID`.

**Release Note**
```release-note
NONE
```

### CODE REVIEW

File: pkg/cmd/operator/operator.go [hunk 1]
```diff
@@ -243,19 +242,20 @@ func Run(healthPort, monitoringPort int32, leaderElection bool, leaderElectionID
...
+		platformName = operatorID
 	} else {
 		platformName = platform.DefaultPlatformName
 	}
 
 	if pl, err := kubernetes.GetIntegrationPlatform(ctx, c, platformName, operatorNamespace); pl == nil || k8serrors.IsNotFound(err) {
 		defaultPlatform := v1.NewIntegrationPlatform(operatorNamespace, platformName)
-		if defaultPlatform.Labels == nil {
-			defaultPlatform.Labels = make(map[string]string)
+		v1.SetAnnotation(&defaultPlatform.ObjectMeta, "camel.apache.org/platform.generated", "true")
```
Comment (christophd) [inline pkg/cmd/operator/operator.go:255]: This sets an annotation, before this has been set as a label. I have no idea what implications this change will have

**Review by christophd — COMMENTED** [review]
Awesome! many thanks for reporting, analyzing and fixing. 

I only have one doubt on setting `camel.apache.org/platform.generated` as an annotation instead of using a label as it has been before

**Review by squakez — CHANGES_REQUESTED** [review]

File: pkg/cmd/operator/operator.go (hunk 1 above)
Comment (valdar) [reply pkg/cmd/operator/operator.go]: Changed back to be a label.

**Review by squakez — APPROVED** [review]
LGTM, Thanks!

File: pkg/cmd/operator/operator.go (hunk 1 above)
Comment (valdar) [reply pkg/cmd/operator/operator.go]: Reverted to label.

**Review by christophd — APPROVED** [review]
LGTM! thanks

**Comment by christophd** [comment]
+1 on backporting to 1.12 and 1.10

**Comment by valdar** [comment]
> @valdar please consider if it makes sense to backport to 1.12 and 1.10 LTS as well.

I think it does, will do that.

**Pull request merged**